src/assets/scripts/client/airport/MapModel.js

```diff
@@ -0,0 +1,135 @@
+import _forEach from 'lodash/forEach';
+import _isEqual from 'lodash/isEqual';
+import _isNumber from 'lodash/isNumber';
+import _map from 'lodash/map';
+import BaseModel from '../base/BaseModel';
+import DynamicPositionModel from '../base/DynamicPositionModel';
+import StaticPositionModel from '../base/StaticPositionModel';
+
+/**
+ * A video map item, containing a collection of map lines
+ *
+ * Defines a vide omap layer referenced by an `AirportModel` that
+ * contains the map lines to be drawn by the `CanvasController`,
+ * as well as a name describing the map contents and a flag
+ * allowing rendering of the layer to be suppressed.
+ *
+ * @class MapModel
+ */
+export default class MapModel extends BaseModel {
+    /**
+     * @for MapModel
+     * @constructor
+     * @param map {object}
+     * @param airportPosition {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(map, airportPosition, magneticNorth) {
+        super();
+
+        if (!map || !airportPosition || !_isNumber(magneticNorth)) {
+            throw new TypeError(
+                'Invalid parameter, expected map, airportPosition and magneticNorth to be defined'
+            );
+        }
+
+        /**
+         * A flag indicating whether the layer should be hidden
+         *
+         * @for MapModel
+         * @property isHidden
+         * @type {boolean}
+         * @default false
+         */
+        this.isHidden = false;
```

Comment:
So here's where I'd like us to make a small fundamental change... Instead of setting whether or not the map is "hidden" in the definition of the map _itself_, which maps are shown/hidden should be determined in a singular place in the airport file, called `defaultMaps`. This would be an array of strings which match the map names.

Then, during the initialization of the MapModel, we will set the `.isHidden` property based on whether or not `airportModel.defaultMaps.includes(mapModel.name)`. This could be passed down during instantiation if needed, or accessed at the appropriate time.

This way, when the maps are super long, it's easy for us to go into the airport file and quickly change which maps to display on initial load. When turning maps on/off during the simulation, the maps can still be hidden/unhidden as necessary.

Reply:
> This way, when the maps are super long
Yup, that struck me when having to fold the JSON. That should be relatively simple to implement and test for.